minigalaxy/ui/gametile.py

```diff
@@ -180,10 +188,18 @@ def __download_file(self) -> None:
         finish_func = self.__install
         number_of_files = len(download_info['files'])
         for key, file_info in enumerate(download_info['files']):
-            if key > 0:
-                download_path = "{}-{}.bin".format(self.download_path, key)
+            download_url = self.api.get_real_download_link(file_info["downlink"])
+            try:
+                # Extract the filename from the download url (filename is between %2F and &token)
+                download_path = os.path.join(self.download_dir, urllib.parse.unquote(re.search('%2F(((?!%2F).)*)&t', download_url).group(1)))
+                if key == 0:
+                    # Iff key = 0, denote the file as the executable's path
+                    self.download_path = download_path
+            except AttributeError:
```

Comment:
When would this exception trigger?

Reply:
It would trigger in case the regular expression can't find any matching string in the link, so `re.search()` returns None and `<None>.group(1)` triggers the exception

Reply:
Ah okay, that makes sense. Thanks!